### DISCUSSION TITLE
Failing to run LIS with GFS forcing

### DISCUSSION THREAD

**Discussion by fepeacock** [NASA-LIS/LISF#1391] [Q&A] [answered]
Hello,

I am new to LISF. I am using LISF-x86_64.AppImage (LISF Public 7.4.2 Release on Ubuntu 22.04). 

I am able to successfully complete testcases 1 and 2 using a download script to download the NLDAS2 data.

I am failing to use GFS data. I read the F90 files in the lis/metforcing/gfs directory and it appears that the files should have the following filename template: YYYYMMDDCC.gfs.sfluxgrbfFF.sg where CC is the cycle and FF is the forecast hour. 

I am surprised that the extension is .sg since the downloaded files from the NOMADS server ends in .grib2. I am also surprised that in the create_gfsfilename.F90 file defines the fsubs as being 21 characters long but the subsequent creation of the variable is only 19 characters.

I renamed NOMADS files with filename template: gfs.t00z.sfluxgrbf003.grib2 to the above filename template based on the cycle and forecast hour and option 1 (bookend 1).

The error I am getting ends with: (the full lislog file is included in the attached zip file)
 Using GFS forcing
 GFS forcing directory :./INPUT/GFS.FORCING
 [INFO] Reading Noah-3.6 vegetation parameter file: ./INPUT/LS_PARAMETERS/noah36_parms/VEGPARM.TBL
 [INFO] Noah-3.6 Landuse type MODI - found           20  categories
 [INFO] Reading Noah-3.6 greenness fraction from greenness maps
 [INFO] Reading SHDMAX map from ./lis_input.uk_1modeltype.noah36.d01.nc
 [INFO] Reading SHDMIN map from ./lis_input.uk_1modeltype.noah36.d01.nc
 [INFO] Reading Noah-3.6 albedo from albedo maps
 [INFO] Calculating Noah-3.6 LAI from greenness and LAI min/max in table
 [INFO] Reading Noah-3.6 max snow albedo from max snow albedo maps
 [INFO] noah36_setsoils -- reading soil files
 [INFO] Fixing Noah-3.6 slope index to type:            1
 [INFO] Reading Noah-3.6 soil parameter file: ./INPUT/LS_PARAMETERS/noah36_parms/SOILPARM.TBL.WRF-3.9
 [INFO] Reading Noah-3.6 general parameter file: ./INPUT/LS_PARAMETERS/noah36_parms/GENPARM.TBL
 [INFO] noah36_coldstart -- cold-starting Noah-3.6
 [INFO] noah36_coldstart -- Using the specified start time    2023.5424657534247
 [INFO]: LIS timestep    900.000000
[INFO] LIS cycle time: 07/18/2023 00:15:00
 [INFO] Reading greenness map for month            7 from ./lis_input.uk_1modeltype.noah36.d01.nc
 [INFO] Reading greenness map for month            8 from ./lis_input.uk_1modeltype.noah36.d01.nc
 [INFO] Reading albedo map for month            7
 [INFO] Reading albedo map for month            8
 MSG: get_gfs -- changing grid to 20150114 --
 Getting new time1 data
 ERROR: GFS data gap exceeds 10 days on file 1
 [ERR] endrun is being called

I believe the problem is a file naming error as my ./INPUT/GFS.FORCING/202307/ directory has the following files: 
2023071618.gfs.sfluxgrbf00.sg 2023071700.gfs.sfluxgrbf00.sg

If someone could assist me, I would be extremely grateful.

Frank
[LIS_Files.zip](https://github.com/NASA-LIS/LISF/files/12208704/LIS_Files.zip)

**Comment by dmocko** [comment, marked as answer]
Hi @fepeacock 

Unfortunately, the GFS metforcing plugin in LIS is a bit outdated, and simply re-naming the newer GRIB-2 files from NOMADS won't work for you (as you have seen).  The metforcing code will need to be re-written to handle the updated format.:
https://github.com/NASA-LIS/LISF/tree/master/lis/metforcing/gfs

We welcome code contributions, if you would like to re-write this code to handle the updated format.  Please see: https://github.com/NASA-LIS/LISF/blob/master/CONTRIBUTING.md

You can use code here as a guide to how to read in GRIB-2 data:
https://github.com/NASA-LIS/LISF/tree/master/lis/metforcing/gefs
https://github.com/NASA-LIS/LISF/blob/master/lis/metforcing/usaf/AGRMET_fldbld_gfs.F90
https://github.com/NASA-LIS/LISF/blob/master/lis/metforcing/usaf/AGRMET_fldbld_precip_gfs.F90
